llvm/utils/UpdateTestChecks/common.py

```diff
@@ -437,10 +500,22 @@ def applySubstitutions(s, substitutions):
     return s
 
 
+def applyArgArraySubstitutions(args: list, substitutions):
+    new_args = []
+
+    # After expanding substitutions, args might be turned into multiple ones.
+    # Make sure to re-split.
+    for arg in args:
+        new_args += shlex.split(applySubstitutions(arg, substitutions))
+    return new_args
```

Comment:
Nit (?): Alternatively, this could also be a map or list comprehension instead. Something on the lines of `[shlex.split(applySubstitions(args, substitution)) for arg in `args]`` or `args.map(lambda arg: ...)`

Reply:
I have changed that bit in follow-up fixups to support Windows as well, maybe it's a bit more readable.